src/iface/fragments.rs

```diff
@@ -0,0 +1,170 @@
+use storage::Assembler;
+use time::Instant;
+use managed::ManagedSlice;
+
+/// Default timeout duration
+const FRAGMENTATION_TIMEOUT_MS: i64 = 500;
+
+#[allow(dead_code)]
+pub struct Packet<'a, T: 'a> {
+    assembler: Assembler,
+    pub rx_buffer: ManagedSlice<'a, T>,
+    id: u16,
+    last_used: Instant,
+    header: bool,
+    header_len: usize,
+    total_len: usize,
+}
+
+
+impl<'a, T: 'a> Packet<'a, T> {
+    pub fn new<S>(storage: S) -> Packet<'a, T>
+	    where S: Into<ManagedSlice<'a, T>>,
+    {
+    	let s = storage.into();
+        Packet {
+            assembler: Assembler::new(s.len()),
+            rx_buffer: s,
+            id: 0,
+            last_used: Instant::from_millis(0),
+            header: false,
+            header_len: 0,
+            total_len: 0,
+        }
+    }
+
+    /// Reset packet
+    pub fn reset(&mut self) {
```

Comment:
I think `Packet` would be much cleaner if instead of these empty values and `set_id` it was defined like...
```
struct Packet {
  rx_buffer: ManagedSlice<'a, T>,
  state: PacketState  
}

enum PacketState {
  Empty,
  Assembling {
    assembler: ...,
    has_header: ...,
    etc
  }
}
```
and then instead of methods like `set_id` you would have methods for transition between states, i.e. `start(ident: u16)`, `add_fragment(...)`, `extract_packet()`, and calling a method in the state that isn't correct for it would panic. Currently, this class of logic errors is not guarded against at all.

Reply:
Done.